### PR TITLE
Cypress

### PR DESCRIPTION
**Pull request by DevAlanex**

### CODE REVIEW

File: cypress/e2e/spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    cy.visit('http://localhost:8080/')
+  })
+})
+
+describe('Chart Builder', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:8080/line.html');
+    
+      // Filling in label inputs
+      cy.get('#chart-title-input').type('Cats vs. Dogs');
```
Comment (sourcewire) [inline cypress/e2e/spec.cy.js]: I see you're finding elements by their id. With cypress testing we're supposed to be navigating the site like how a user would. Users don't see 'id=chart-title-input'. They see 'Chart title'. Worried we might get knocked down points for this.

File: cypress/e2e/spec.cy.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  });
+
+  it('saves', () => {
+
+    // Assertion
+    cy.get('#generate-chart-btn').click();
+    cy.get('#chart-display').should('not.be.empty');
+
+    cy.get('#save-chart-btn').click();
+    cy.visit('http://localhost:8080/');
```
Comment (sourcewire) [inline cypress/e2e/spec.cy.js]: It look like this creates and saves a chart, then navigates to the gallery. But it doesn't actually verify the chart we just made is actually in the gallery. Unless I'm missing something...

File: cypress/e2e/spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    cy.get('#x-y-data-grid').find('input').should('have.length', 4); // cleared
+  });
+
+  it('generation', () => {
+
+    // Assertion
+    cy.get('#generate-chart-btn').click();
+    cy.get('#chart-display').should('not.be.empty');
+
+    cy.get('#x-y-data-grid').find('input').should('have.length', 14);
```
Comment (sourcewire) [inline cypress/e2e/spec.cy.js]: can you explain this?

**Review by sourcewire — CHANGES_REQUESTED** [review]
I have a couple questions. Test names are also kinda vague and no comments either. Looks good though.

File: cypress/e2e/spec.cy.js (hunk 1 above)
Comment (baumannevan) [reply cypress/e2e/spec.cy.js]: image
I would if at all possilble try to switch to using the find-by-role or label text if possible. if ids are necessary i would switch them to a testID

File: cypress/e2e/spec.cy.js (hunk 3 above)
Comment (baumannevan) [reply cypress/e2e/spec.cy.js]: the additon of comments would be helpful. i don't know if it was in the requirement for the final but i think well documented was something we where scored on for all the other projects

**Review by baumannevan — APPROVED** [review]
Functionally looks pretty good. The additon of comments would be nice so others can quickly understand what is happening in each test, and in each AAA phase.

File: cypress/e2e/spec.cy.js
Comment (baumannevan) [inline cypress/e2e/spec.cy.js:1]: thanks for changing to fit user interactions. the code looks much better and is easier to understand with the comments you added. big thanks great job!!

**Review by baumannevan — APPROVED** [review]
looks good. concers addressed, big win

**Review by sourcewire — APPROVED** [review]
Great work thanks for the changes! One thing, it looks like the features 'Saving a chart to the “gallery”' and 'Re-opening a saved chart' are combined into one test. It might be worth splitting those into two tests.

File: cypress/e2e/spec.cy.js (hunk 1 above)
Comment (sourcewire) [reply cypress/e2e/spec.cy.js]: great work!

**Pull request merged**